source/infrastructure/lib/core-scheduler.ts

```diff
@@ -0,0 +1,159 @@
+    cfnStateTable.overrideLogicalId('StateTable')
+    cfnStateTable.addPropertyOverride("SSESpecification", {
+      "KMSMasterKeyId": { "Ref": "InstanceSchedulerEncryptionKey" },
+      "SSEEnabled": true,
+      "SSEType": 'KMS'
+    })
+
+    this.configTable = new dynamodb.Table(scope, 'ConfigTable', {
+      sortKey: {
+        name: 'name',
+        type: dynamodb.AttributeType.STRING
+      },
+      partitionKey: {
+        name: 'type',
+        type: dynamodb.AttributeType.STRING
+      },
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+      removalPolicy: RemovalPolicy.DESTROY,
+      pointInTimeRecovery: true
+    })
+
+    const cfnConfigTable = this.configTable.node.defaultChild as dynamodb.CfnTable
+    cfnConfigTable.overrideLogicalId('ConfigTable')
+    cfnConfigTable.addPropertyOverride("SSESpecification", {
+      "KMSMasterKeyId": { "Ref": "InstanceSchedulerEncryptionKey" },
+      "SSEEnabled": true,
+      "SSEType": 'KMS'
+    })
+
+    this.maintenanceWindowTable = new dynamodb.Table(scope, 'MaintenanceWindowTable', {
+      partitionKey: {
+        name: 'Name',
+        type: dynamodb.AttributeType.STRING
+      },
+      sortKey: {
+        name: "account-region",
+        type: dynamodb.AttributeType.STRING
+      },
+      billingMode: dynamodb.BillingMode.PAY_PER_REQUEST,
+      removalPolicy: RemovalPolicy.DESTROY,
+      pointInTimeRecovery: true
+    })
+
+    const cfnMaintenanceWindowTable = this.maintenanceWindowTable.node.defaultChild as dynamodb.CfnTable
+    cfnMaintenanceWindowTable.overrideLogicalId('MaintenanceWindowTable')
+    cfnMaintenanceWindowTable.addPropertyOverride("SSESpecification", {
+      "KMSMasterKeyId": { "Ref": "InstanceSchedulerEncryptionKey" },
+      "SSEEnabled": true,
+      "SSEType": 'KMS'
+    })
+
+    this.lambdaFunction.addEnvironment('CONFIG_TABLE', cfnConfigTable.ref)
+    this.lambdaFunction.addEnvironment('MAINTENANCE_WINDOW_TABLE', cfnMaintenanceWindowTable.ref)
+    this.lambdaFunction.addEnvironment('STATE_TABLE', cfnStateTable.ref)
+
+
+    const dynamodbPolicy = new iam.PolicyStatement({
+      actions: [
+        'dynamodb:DeleteItem',
+        'dynamodb:GetItem',
+        'dynamodb:PutItem',
+        'dynamodb:Query',
+        'dynamodb:Scan',
+        'dynamodb:BatchWriteItem'
+      ],
+      effect: iam.Effect.ALLOW,
+      resources: [
+        cfnConfigTable.attrArn,
+        cfnMaintenanceWindowTable.attrArn
+      ]
+    })
+
+    this.lambdaFunction.addToRolePolicy(dynamodbPolicy)
+
+    NagSuppressions.addResourceSuppressions(lambdaToDynamoDbConstruct.node.findChild("LambdaFunctionServiceRole"), [{
+      id: "AwsSolutions-IAM5",
+      reason: "This Lambda function needs to be able to write a log streams for each scheduler execution (1 per account/region/service)"
+    }])
+  }
+
+  private lambdaToDynamoPropsFrom(props: InstanceSchedulerLambdaProps) : LambdaToDynamoDBProps {
```

Comment:
I don't think this needs to be a member function. Also, the name confuses me, but I'm not sure what would be better because of the overloaded terms "construct" and "props".

Reply:
renamed to extractLambdaToDynamoPropsFrom(). 

As for the pros/cons of having the function at all. I feel it allows the code to more clearly document exactly what is happening. I can look at the high-level resources being created and not get distracted by a huge parameter list that would otherwise distract from what's actually going on